### PR TITLE
codecoverage/backend: Serve history with overall coverage

### LINKED ISSUE #1226: Add API to retrieve list of changesets with overall coverage
This is going to be needed by the frontend in order to build a graph showing the evolution of the coverage.

### PR DESCRIPTION
**Pull request by La0**
Fixes #1226.

### CODE REVIEW

File: src/codecoverage/backend/codecoverage_backend/services/gcp.py [hunk 1]
```diff
@@ -249,3 +271,44 @@ def get_coverage(self, repository, changeset, path):
...
+
+    def get_history(self, repository, path='', start=None, end=None):
+        '''
+        Load the history overall coverage from the redis cache
+        Default to date range from now back to a year
+        '''
+        if end is None:
+            end = calendar.timegm(datetime.utcnow().timetuple())
+        if start is None:
+            start = end - 365*24*3600
```
Comment (marco-c) [inline src/codecoverage/backend/codecoverage_backend/services/gcp.py]: Nit: use relativedelta

File: src/codecoverage/backend/codecoverage_backend/services/gcp.py [hunk 2]
```diff
@@ -249,3 +271,44 @@ def get_coverage(self, repository, changeset, path):
...
+        )
+
+        def _coverage(changeset, date):
+            # Load overall coverage for specified path
+            changeset = changeset.decode('utf-8')
+            key = KEY_OVERALL_COVERAGE.format(
+                repository=repository,
+                changeset=changeset,
+            )
+            coverage = self.redis.hget(key, path)
```
Comment (marco-c) [inline src/codecoverage/backend/codecoverage_backend/services/gcp.py]: Will this be fast enough? We'll call hget thousand of times

Comment (La0) [reply src/codecoverage/backend/codecoverage_backend/services/gcp.py]: As discussed on IRC, we have several options to optimize later on:
1. Use redis pipelining to batch requests
2. Store data more efficiently for retrieval (`overall:<repo>:<path>` :arrow_right: `[0.1, 0.2, ...]`
3. Build dedicated reports on the bot

Comment (marco-c) [reply src/codecoverage/backend/codecoverage_backend/services/gcp.py]: Want to file an issue with these options? (call it "Consider optimizing history ...")

**Pull request merged**